### ISSUE TITLE
[MCOMPILER-311] NPE when --patch-module is used

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-compiler-plugin#514]
**[Zhihong Zhang](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zhihong)** opened **[MCOMPILER-311](https://issues.apache.org/jira/browse/MCOMPILER-311?redirect=false)** and commented

When "--patch-module" argument is added to the compiler, the plugin always throws NullPointerException. For example,

```xml
<compilerArgs>
    <arg>--patch-module</arg>
    <arg>java.xml.ws.annotation=jsr305-3.0.1.jar</arg>
</compilerArgs>
```

```
Caused by: java.lang.NullPointerException
        at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1004)
        at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:168)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
```

The bug is caused by an unimplemented method in CompilerMOJO.java,

```
@Override
protected Map<String, JavaModuleDescriptor> getPathElements()
{
    // TODO Auto-generated method stub
    return null;
}
```

---

**Affects:** 3.7.0

**Comment by jira-importer** [comment]
**[Anthony Baldocchi](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=abaldocchi)** commented

This syntax works for me:

```
<arg>--patch-module=java.xml.ws.annotation=jsr305-3.0.1.jar</arg>
```

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [e258b6572a31496c8a10cd3d3d035b2da12a343e](https://gitbox.apache.org/repos/asf?p=maven-compiler-plugin.git;a=commit;h=e258b6572a31496c8a10cd3d3d035b2da12a343e)
